apps/entity-service/src/entities/processors/media.processor.ts

```diff
@@ -0,0 +1,245 @@
+        where: {
+          [Op.and]: [
+            { siteId: { [Op.in]: siteSubquery.literal } },
+            Sequelize.where(fn("MONTH", col("due_at")), projectReportWithDue.dueAt.getMonth() + 1),
+            Sequelize.where(fn("YEAR", col("due_at")), projectReportWithDue.dueAt.getFullYear())
+          ]
+        },
+        attributes: ["id"]
+      });
+    }
+
+    models.push({ modelType: SiteReport.LARAVEL_TYPE, subquery: siteReports.map(report => report.id) });
+
+    let nurseryReports = [];
+    if (projectReportWithDue.dueAt) {
+      nurseryReports = await NurseryReport.findAll({
+        where: {
+          [Op.and]: [
+            { nurseryId: { [Op.in]: nurserySubquery.literal } },
+            Sequelize.where(fn("MONTH", col("due_at")), projectReportWithDue.dueAt.getMonth() + 1),
+            Sequelize.where(fn("YEAR", col("due_at")), projectReportWithDue.dueAt.getFullYear())
+          ]
+        },
+        attributes: ["id"]
+      });
+    }
+    models.push({ modelType: NurseryReport.LARAVEL_TYPE, subquery: nurseryReports.map(report => report.id) });
+
+    return models;
+  }
+
+  private async getSiteReportModels(siteReport: SiteReport) {
+    return [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [siteReport.id] }];
+  }
+
+  private async buildQuery(baseEntity: EntityModel, query: MediaQueryDto) {
+    const userAssociations: Includeable = {
+      association: "createdByUser",
+      attributes: ["firstName", "lastName"]
+    };
+
+    const builder = await this.entitiesService.buildQuery(Media, query, [userAssociations]);
+
+    let models = [];
+    if (baseEntity instanceof Project) {
+      models = await this.getProjectModels(baseEntity);
+    } else if (baseEntity instanceof Site) {
+      models = await this.getSiteModels(baseEntity);
+    } else if (baseEntity instanceof Nursery) {
+      models = await this.getNurseryModels(baseEntity);
+    } else if (baseEntity instanceof ProjectReport) {
+      models = await this.getProjectReportModels(baseEntity);
+    } else if (baseEntity instanceof SiteReport) {
+      models = await this.getSiteReportModels(baseEntity);
+    } else if (baseEntity instanceof NurseryReport) {
+      models = [{ modelType: this.entityModelClass.LARAVEL_TYPE, subquery: [baseEntity.id] }];
+    }
+
+    builder.where({
+      [Op.or]: models.map(model => {
+        return {
+          modelType: model.modelType,
+          modelId: {
+            [Op.in]: model.subquery
+          }
+        };
+      })
+    });
+
+    if (query.isGeotagged != null) {
+      builder.where({
+        [Op.and]: [
+          {
+            lat: {
+              [query.isGeotagged ? Op.ne : Op.eq]: null
+            },
+            lng: {
+              [query.isGeotagged ? Op.ne : Op.eq]: null
+            }
+          }
+        ]
+      });
+    }
+
+    if (query.isPublic != null) {
+      builder.where({
+        isPublic: query.isPublic ? "1" : "0"
```

Comment:
`isPublic` on the model is boolean, this should simply be `isPublic: query.isPublic`.

Reply:
fixed